src/kademlia/get_item.cpp

```diff
@@ -88,39 +91,27 @@ void get_item::got_data(bdecode_node const& v,
 		// and it's immutable
 
 		m_data.assign(v);
-		bool put_requested = m_data_callback(m_data, true);
 
-		// if we intend to put, we need to keep going
-		// until we find the closest nodes, since those
-		// are the ones we're putting to
-		if (put_requested)
-		{
+		// There can only be one true immutable item with a given id
```

Comment:
is it a safe assumption that we're not putting a mutable blob here? just because I asked for a mutable blob doesn't mean the other end can't send me a response that looks like an immutable one. That shouldn't make me abort though, right? or am I misreading this?

It looks like we make an assumption about what _we_ are doing and want to do based on what an untrusted peer told us.

Another powerful feature that may be lost here is that the client can choose whether to update the blob or not _after_ having seen it. Perhaps it doesn't need any updating, it's already at the latest version. We won't know that until we read it from the DHT.